README.md

```diff
@@ -1,5 +1,278 @@
+> "I am inevitable!"
+>           - A mean purple titan
+
 # Overview
 
-Library for constructing structured text.
+Constructor library for structured monospaced text.
+
 Current functionality:
-* Pending...
+* Multi-line text of uniform character width, and arbitrary padding
+* Multi-line columnated-row text with vertical dividers
+* Textual table with horizontal dividers, header, and footer
+* Blocked text with border, title, and caption
+
+Each of these elements can be combined to create structured text that can be printed to a console, or used to display arbitrarily sized text data to the screen.
+
+*Note*: The display font must render with a monospaced font for the text to align as intended.
+
+## Kotlin & Java Compatibility
+
+This library was constructed using Kotlin, and as such used several idiomatic Kotlin structures for the API. However, it was also designed to emulate a standard Java builder pattern structure for the API. So each construct can be called using either approach.
+
+### Target Compatibility
+
+|        | Version |
+|--------|---------|
+| JVM    | `1.9`   |
+| Kotlin | `1.3.x` |
+
+## Usage Examples
+
+*Note*: These examples can be composed with each other in arbitrary ways to render the structured text.
+
+### Multi-line Uniform Text
+
+#### Kotlin API
+
+Let the pre-constructed line values be:
+```kotlin
+val line1 = faker.getWords(5)
+val line2 = faker.getWords(2)
+val line3 = faker.getWords(3)
+```
+
+Then, these values can be used in the following ways:
+
+##### Option 1
+```kotlin
+textCell(line1, line2, line3) {
```

Comment:
could place all kotlin options in the same block:
```kotlin
textCell(blah) {

}
// or
textCell {
withBlah
}
// or
textCell {
}
```

Reply:
I'm puzzled what you're suggesting here. You don't think I should enumerate the options in separate blocks? Doesn't make a difference to me, but if you think it's clearer I'll update.

Reply:
Well, looking at the readme now I think in separate blocks is fine.

Reply:
I'm fine with either way. I'm not sure which is less confusing. I'm just trying to give examples of how to call, not necessarily enumerate all possible ways. Maybe without the blocks conveys that better?

Reply:
Yeah without blocks conveys that